_posts/2022-04-01-tidying-snapshots-up.md

```diff
@@ -0,0 +1,213 @@
+snapshot. You can not do this earlier, because without any completed checkpoints Flink will - upon
+failure - try to recover from the initial snapshot.
+
+<div style="text-align: center">
+  <img src="{{ site.baseurl }}/img/blog/2022-04-xx-tidying-snapshots/restore-mode-no_claim.svg" alt="NO_CLAIM restore mode" width="70%" >
+</div>
+
+If you do not want to sacrifice any performance while taking the first checkpoint, we suggest
+looking into the `CLAIM` mode.
+
+## CLAIM mode
+
+The other available mode is the *CLAIM* mode. In this mode Flink claims ownership of the snapshot
+and essentially treats it like a checkpoint: it controls the lifecycle and might delete it if it is
+not needed for recovery anymore. Hence, it is not safe to manually delete the snapshot or to start
+two jobs from the same snapshot. Flink keeps around a configured number of checkpoints.
+
+<div style="text-align: center">
+  <img src="{{ site.baseurl }}/img/blog/2022-04-xx-tidying-snapshots/restore-mode-claim.svg" alt="CLAIM restore mode" width="70%">
+</div>
+
+Each restore mode serves a specific purpose. Still, we believe the default *NO_CLAIM* mode is a good
+tradeoff in most situations, as it provides clear ownership with a small price for the first
+checkpoint after the restore.
+
+You can pass the restore mode as:
+
+```
+$ bin/flink run -s :savepointPath -restoreMode :mode -n [:runArgs]
+```
+
+# Savepoint format
+
+The other, already mentioned improvement, is the possibility to trigger savepoints in state backends
+native format. This has been introduced to match two characteristics, one of both savepoints and
+checkpoints:
+
+- being self-contained, relocatable, and owned by users
+- lightweight and thus fast to take and recover from
+
+In order to provide the two features in a single concept we provided a way for Flink to create a
+savepoint in a binary format of the used state backend. This brings a significant difference
+primarily in combination with the `state.backend: rocksdb` and incremental snapshots.
+
+That state backend can leverage RocksDB native on-disk data structures which are usually referred to
+as SST files. Incremental checkpoints leveraged those files. Basically incremental snapshots are
+collections of those SST files with some additional metadata, which can be rather quickly reloaded
+into the working directory of RocksDB upon restore.
+
+Native savepoints can use the same mechanism of uploading the SST files instead of dumping the
+entire state into a canonical Flink format. There is one additional benefit over simply using the
+externalized incremental checkpoints that native savepoints are still relocatable and self-contained
+in a single directory. In case of checkpoints that does not hold, because a single SST file can be
+used by multiple checkpoints, and thus is put into a common shared directory. That is why they are
+called incremental.
+
+You can choose the savepoint format when triggering the savepoint:
```

Comment:
```suggestion
## Savepoint format

You can now trigger savepoints in the native format of state backends.
This has been introduced to match two characteristics, one of both savepoints and
checkpoints:

- self-contained, relocatable, and owned by users
- lightweight (and thus fast to take and recover from)

In order to provide the two features in a single concept, we provided a way for Flink to create a
savepoint in a (native) binary format of the used state backend. This brings a significant difference
especially in combination with the `state.backend: rocksdb` setting and incremental snapshots.

That state backend can leverage RocksDB native on-disk data structures which are usually referred to
as SST files. Incremental checkpoints leveraged those files and are
collections of those SST files with some additional metadata, which can be quickly reloaded
into the working directory of RocksDB upon restore.

Native savepoints can use the same mechanism of uploading the SST files instead of dumping the
entire state into a canonical Flink format. There is one additional benefit over simply using the
externalized incremental checkpoints: native savepoints are still relocatable and self-contained
in a single directory. In case of checkpoints that do not hold, because a single SST file can be
used by multiple checkpoints, and thus is put into a common shared directory. That is why they are
called incremental.

You can choose the savepoint format when triggering the savepoint like this:
```